### PR TITLE
SDK_VERSION: add a current-sdk-version.txt file and use it for documentation and workflows

### PR DESCRIPTION
**Pull request by fabiobaltieri**
This is meant to be used for https://github.com/zephyrproject-rtos/zephyr/issues/64926 to remove the hardcoded SDK version and have an "auto" option that gets the current one in the Zephyr checkout automatically.

---

- add a `SDK_VERSION` top level file with the current SDK version
- replace all documentation occurrences with a set of automatic substitutions
  - https://builds.zephyrproject.io/zephyr/pr/67603/docs/develop/getting_started/installation_linux.html
  - https://builds.zephyrproject.io/zephyr/pr/67603/docs/develop/toolchains/zephyr_sdk.html
- set `ZEPHYR_SDK_INSTALL_DIR` automatically in the workflow specs
  - https://github.com/zephyrproject-rtos/zephyr-testing/tree/vfabio-testing-branch

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Is this a reasonable location? Was thinking about keeping it close to the other workflow just for convenience but it could go anywhere.

**Review by stephanosio — CHANGES_REQUESTED** [review]
This will need to be accompanied by a complete solution replacing all hard-coded Zephyr SDK version numbers throughout the repository (i.e. in the GitHub workflows as well as in the docs).

Considering the documentation case, it would be better to completely decouple the "recommended SDK version number" from the GitHub Actions -- for example, we could introduce something like `SDK_VERSION` under the Zephyr repository root that specifies the "recommended Zephyr SDK version" for a particular Zephyr release.

**Review by gmarull — COMMENTED** [review]
note sure if github workflows is the right folder, why not a top-level file named "SDK-VERSION"?

**Comment by fabiobaltieri** [comment]
Top level `SDK_VERSION` sounds good to me

> This will need to be accompanied by a complete solution replacing all hard-coded Zephyr SDK version numbers throughout the repository (i.e. in the GitHub workflows as well as in the docs).
> 
> Considering the documentation case, it would be better to completely decouple the "recommended SDK version number" from the GitHub Actions

Cool, yeah the actions could read the file and put it back in the environment and the docs could generate a substitution from it. The two statements seems to be in contrast though, are you suggesting that a single file should update actions and documentation (which sounds reasonable to me) or that they should be decoupled and use two different files or something?

**Comment by stephanosio** [comment]
> are you suggesting that a single file should update actions and documentation

Single top-level `SDK_VERSION` for both GHA and docs.

**Comment by nashif** [comment]
> Single top-level `SDK_VERSION` for both GHA and docs.

+1

**Comment by fabiobaltieri** [comment]
Alright all references to 0.16.4 apart from the new file are gone.

Had to change the documentation blocks from `code-block` to `parsed-literal` and add way more substitutions that I originally thought to make URL parse correctly but it works now.

The workflows have been updated as well, most are using the current config but the codesize one is not and it [seems to work fine](https://github.com/zephyrproject-rtos/zephyr/actions/runs/7532844141/job/20504235633?pr=67603), plus I pushed it on the [testing repo](https://github.com/zephyrproject-rtos/zephyr-testing/tree/vfabio-testing-branch).

**Review by nashif — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Review by stephanosio — COMMENTED** [review]
The changes themselves look good to me; but, the first commit title seems a little off.

> [SDK_VERSION: add a **current-sdk-version.txt** file](https://github.com/zephyrproject-rtos/zephyr/pull/67603/commits/55527fd98666f5d57bd8a53880c587eebb85e981)

**Review by stephanosio — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]
Very nice, thanks Fabio!

File: doc/develop/getting_started/installation_linux.rst [hunk 1]
```diff
@@ -227,32 +227,31 @@ The Zephyr SDK supports the following target architectures:
 
 Follow these steps to install the Zephyr SDK:
 
-#. Download and verify the `Zephyr SDK bundle
-   <https://github.com/zephyrproject-rtos/sdk-ng/releases/tag/v0.16.4>`_:
+#. Download and verify the `Zephyr SDK bundle`_:
 
-   .. code-block:: bash
+   .. parsed-literal::
```
Comment (kartben) [inline doc/develop/getting_started/installation_linux.rst:232]: Nice find! I actually wanted to implement substitutions for the SDK version a while back but couldn't find a way to have them work in code-blocks :)
One thing that I'd like to see is that the block still renders the same as other code blocks (ex. it should have a background ; ideally, syntax highlighting too, I guess) 

image

Comment (kartben) [reply doc/develop/getting_started/installation_linux.rst:232]: https://github.com/sphinx-doc/sphinx/issues/3306

Comment (fabiobaltieri) [reply doc/develop/getting_started/installation_linux.rst:232]: Yeah, it's a bit of a compromise, I've figured you'd probably follow up with a strategic CSS change.

Comment (kartben) [reply doc/develop/getting_started/installation_linux.rst:232]: I don't want to block the rest of this PR, but would it make sense to keep the doc update for later? Ideally you'd open a separate PR with the doc commit there that we (I) can tackle in due course. Beyond the (arguably not critical) issue described in this comment, there is a more concerning one with... wait for it... the PDF file :)

image

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/develop/getting_started/installation_linux.rst (hunk 1 above)
Comment (fabiobaltieri) [reply doc/develop/getting_started/installation_linux.rst:232]: > the PDF file

I HAVE AN IDEA!

Comment (fabiobaltieri) [reply doc/develop/getting_started/installation_linux.rst:232]: ...is there a way of making the `parsed-literal` not highlight? Vaguely remember reading something about it, I'll look a bit and if that's not feasible I can pull the doc change.

Comment (gmarull) [reply doc/develop/getting_started/installation_linux.rst:232]: > > the PDF file
> 
> I HAVE AN IDEA!

PLEASE KILL PDF ONCE AND FOR ALL

**Comment by fabiobaltieri** [comment]
@kartben I've added some backticks to the replaced string, can you give it a try? It removes highlighting on the html version, who knows what'll do in the pdf...

**Comment by fabiobaltieri** [comment]
ha, seems like it worked?
x

**Review by kartben — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**